src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,138 @@
+import './MovieListPage.css';
+import '../../colors.css';
+
+import { useEffect, useState } from 'react';
+
+import GenreSelect from '../GenreSelect/GenreSelect';
+import SortControl from '../SortControl/SortControl';
+import MovieTile from '../MovieTile/MovieTile';
+import EditMovieDialog from '../EditMovieDialog/EditMovieDialog';
+import MovieDetails from '../MovieDetails/MovieDetails';
+import SearchForm from '../SearchForm/SearchForm';
+
+import { getMovies } from '../../services/api.service';
+
+import { Movie } from '../../models/movie';
+
+import { GENRES } from '../../consts';
+
+const MovieListPage = () => {
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [sortCriterion, setSortCriterion] = useState<'releaseDate' | 'title'>('releaseDate');
+    const [activeGenre, setActiveGenre] = useState<string>('all');
+    const [movieList, setMovieList] = useState<Movie[]>([] as Movie[]);
+    const [showDetailContainer, setShowDetailContainer] = useState<boolean>(false);
+    const [selectedMovie, setSelectedMovie] = useState<Movie>({} as Movie);
+    const [showEditDialog, setShowEditDialog] = useState<boolean>(false);
+
+    useEffect(() => {
+        void getMovies(searchQuery, sortCriterion, activeGenre).then((data) => {
+            const formatMovieList: Movie[] = data.data.map((movieApi) => ({
+                id: movieApi.id,
+                movieName: movieApi.title,
+                releaseYear: +movieApi.release_date.split('-')[0],
+                genres: movieApi.genres,
+                voteAverage: movieApi.vote_average,
+                description: movieApi.overview,
+                duration: movieApi.runtime,
+                imageUrl: movieApi.poster_path,
+            }));
+            setMovieList(formatMovieList);
+        });
+    }, [searchQuery, sortCriterion, activeGenre]);
+
+    const searchMovieByName = (name: string): void => {
+        setSearchQuery(name);
+    };
+
+    const showGenreMovies = (selectedGenre: string): void => {
+        setActiveGenre(selectedGenre);
+    };
+    const openDetailInfo = (openMovieId: number): void => {
+        const selectedMovie: Movie = movieList.filter((movie) => movie.id === openMovieId)[0];
+
```

Comment:
Maybe Array.find() suits better here because you have unique identifier:
const selectedMovie: Movie = movieList.find((movie) => movie.id === openMovieId);